File: cpp/pixels.cpp
/////////////////////////////////////////////////////////////////////////////
//
// COMS30121 - pixels.cpp
// TOPIC: access and set individual pixels in an image
//
// Getting-Started-File for OpenCV
// University of Bristol
//
/////////////////////////////////////////////////////////////////////////////

#include <stdio.h>
#include <opencv/cv.h>        //you may need to
#include <opencv/highgui.h>   //adjust import locations
#include <opencv/cxcore.h>    //depending on your machine setup

using namespace cv;

int main() {

  //create a black 256x256, 8bit, 3channel BGR image in a matrix container
  Mat image(256, 256, CV_8UC3, Scalar(0, 0, 0));

  //set pixels to create colour pattern
  for (int y = 0; y < image.rows; y++) //go through all rows (or scanlines)
    for (int x = 0; x < image.cols; x++) { //go through all columns
      image.at<Vec3b>(y, x)[0] = x; //set blue component
      image.at<Vec3b>(y, x)[1] = y; //set green component  
      image.at<Vec3b>(y, x)[2] = 255 - image.at<Vec3b>(y, x)[1]; //set red component
    }

  //construct a window for image display
  namedWindow("Display window", CV_WINDOW_AUTOSIZE);

  //visualise the loaded image in the window
  imshow("Display window", image);

  //wait for a key press until returning from the program
  waitKey(0); 

  //free memory occupied by image 
  image.release();

  return 0;
}

File: pixels.py
################################################
#
# COMS30068 - pixels.py
# TOPIC: create, save and display an image
#
# Getting-Started-File for OpenCV
# University of Bristol
#
################################################

import cv2
import numpy as np

# create a red 256x256, 8bit, 3channel BGR image in a matrix container
image = np.zeros([256,256,3], dtype=np.uint8)

# set pixels to create colour pattern

for y in range(0, image.shape[0]):  # go through all rows (or scanlines)
	for x in range(0, image.shape[1]):  # go through all columns
		image[y, x, 0] = x # set blue component
		image[y, x, 1] = y # set green component  
		image[y, x, 2] = 255 - image[y, x,1] # set red component

# construct a window for image display
namedWindow = 'Display window'

# visualise the loaded image in the window
cv2.imshow(namedWindow, image)

# wait for a key press until returning from the programw
cv2.waitKey(0)

# closing all open windows
cv2.destroyAllWindows()